resharper/resharper-unity/src/Rider/UnityRefreshTracker.cs

```diff
@@ -107,24 +107,31 @@ public class UnityRefreshTracker
             if (solution.GetData(ProjectModelExtensions.ProtocolSolutionKey) == null)
                 return;
 
-            unitySolutionTracker.IsUnityProject.ViewNotNull(lifetime, (lf, args) =>
+            // Rgc.Guarded - beware RIDER-15577
+            myGroupingEvent = solution.Locks.GroupingEvents.CreateEvent(lifetime, "UnityRefresherOnSaveEvent",
```

Comment:
Should we add this under `unitySolutionTracker.IsUnityProjectFolder.AdviseOnce`?
Why do not put all constructor logic under `unitySolutionTracker.IsUnityProjectFolder.AdviseOnce`?